rs/replicated_state/src/canister_snapshots.rs

```diff
@@ -241,6 +241,8 @@ pub struct CanisterSnapshot {
     /// Snapshot of chunked store.
     chunk_store: WasmChunkStore,
     execution_snapshot: ExecutionStateSnapshot,
+    /// Amount of heap delta produced by this snapshot.
+    heap_delta: NumBytes,
```

Comment:
No, it's not a good idea to try to store the heap delta in this struct. The `heap_delta(&self)` function should compute it from the PageMaps instead. 

The issue is that during checkpointing we actually replace the `PageMaps` by equivalent structs that have their heap deltas wiped, as by that point we have all the data on disk. So we are completely unnecessarily overcounting here. Furthermore, to actually implement this field correctly you would have to carry it through to `CanisterSnapshotBits`, which you are not doing atm.

Reply:
Thanks for pointing this out. I suppose I am missing some of the details here. I thought that snapshots are stored during a checkpoint and never change (unlike the canister's main state that keeps changing over time).

In any case, I updated the function to compute the heap delta based on the PageMaps included in a snapshot.